### ISSUE TITLE
Gym picks up provisioning profile from PROVISIONING_PROFILE environment variable

### ISSUE THREAD

**Issue by Blackclaws** [fastlane/fastlane#19243]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
I've debugged this for quite a while. Gym kept setting a provisioning profile on my frameworks and failed the build because of that. The reason turned out that gym picks up a provisioning profile from PROVISIONING_PROFILE and applies that to all bundles that do not have an explicit mapping in export options.

As far as I can tell this is undocumented behavior as per the gym documentation. The feature itself is useful but if you don't know it exists and use the variable name for a different script in the same CI build this is pretty surprising.

##### Command executed
fastlane gym

##### Complete output when running fastlane, including the stack trace and command used
If you need I can provide it

**Comment by lucgrabowski** [comment]
Are you able to archive app from Xcode?
How is signing set up in your framework targets? Is `Automatically manage signing` turned off, `Team` set to `none`?

**Comment by Blackclaws** [comment]
Yes archiving from xcode works, archiving from fastlane also works fine as long as I don't set the PROVISIONING_PROFILE environment variable. 

Signing in framework targets is not set up at all I think because they don't need to be signed

**Comment by lucgrabowski** [comment]
Could you upload logs / Fastfile - at least the parts that are relevant for the issue?

**Comment by Blackclaws** [comment]
Still an issue

**Comment by woshahua** [comment]
still an issue
